librz/core/cdebug.c

```diff
@@ -639,3 +639,255 @@ RZ_API RzCmdStatus rz_core_debug_plugins_print(RzCore *core, RzCmdStateOutput *s
 	rz_cmd_state_output_array_end(state);
 	return RZ_CMD_STATUS_OK;
 }
+
+/* Print out the JSON body for memory maps in the passed map region */
+static void print_debug_map_json(RzDebugMap *map, PJ *pj) {
+	pj_o(pj);
+	if (map->name && *map->name) {
+		pj_ks(pj, "name", map->name);
+	}
+	if (map->file && *map->file) {
+		pj_ks(pj, "file", map->file);
+	}
+	pj_kn(pj, "addr", map->addr);
+	pj_kn(pj, "addr_end", map->addr_end);
+	pj_ks(pj, "type", map->user ? "u" : "s");
+	pj_ks(pj, "perm", rz_str_rwx_i(map->perm));
+	pj_end(pj);
+}
+
+/* Write the memory map header describing the line columns */
+static void print_debug_map_line_header(RzDebug *dbg) {
+	// TODO: Write header to console based on which command is being ran
+}
+
+/* Write a single memory map line to the console */
+static void print_debug_map_line(RzDebug *dbg, RzDebugMap *map, ut64 addr, RzOutputMode mode) {
+	char humansz[8];
+	if (mode == RZ_OUTPUT_MODE_QUIET) { // "dmq"
+		char *name = (map->name && *map->name)
+			? rz_str_newf("%s.%s", map->name, rz_str_rwx_i(map->perm))
+			: rz_str_newf("%08" PFMT64x ".%s", map->addr, rz_str_rwx_i(map->perm));
+		rz_name_filter(name, 0, true);
+		rz_num_units(humansz, sizeof(humansz), map->addr_end - map->addr);
+		dbg->cb_printf("0x%016" PFMT64x " - 0x%016" PFMT64x " %6s %5s %s\n",
```

Comment:
Since this is located in `librz/core` there is no need to use `cb->printf` here, you could use RzCore and RzCons here freely (this is why we asked to move printing in here after all)/.